### PR TITLE
WINC-607: Add support for platform=none in e2e test suite (Cont.)

### PR DESCRIPTION
**Pull request by jrvaldes**
Follow-up to https://github.com/openshift/windows-machine-config-operator/pull/858

### CODE REVIEW

File: test/e2e/network_test.go [hunk 1]
```diff
@@ -233,6 +231,11 @@ func testNorthSouthNetworking(t *testing.T) {
 	testCtx, err := NewTestContext()
 	require.NoError(t, err)
 
+	// Ignore the application ingress load balancer test for None and vSphere platforms as it has to be created manually
+	if testCtx.CloudProvider.GetType() == config.VSpherePlatformType ||
+		testCtx.CloudProvider.GetType() == config.NonePlatformType {
+		t.Skipf("NorthSouthNetworking test is disabled for platform %s", testCtx.CloudProvider.GetType())
```
Comment (jrvaldes) [inline test/e2e/network_test.go:237]: WMCO e2e test [logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/896/pull-ci-openshift-windows-machine-config-operator-master-vsphere-e2e-operator/1489346837620985856/artifacts/vsphere-e2e-operator/windows-e2e-operator-test-with-custom-vxlan-port/build-log.txt):

```
=== RUN   TestWMCO/network/North_south_networking
    network_test.go:237: NorthSouthNetworking test is disabled for platform VSphere
```

**Comment by jrvaldes** [comment]
/test vsphere-e2e-operator

WMCO CSR approver failed to lookup DNS

File: test/e2e/network_test.go [hunk 2]
```diff
@@ -233,6 +231,11 @@ func testNorthSouthNetworking(t *testing.T) {
 	testCtx, err := NewTestContext()
 	require.NoError(t, err)
 
+	// Ignore the application ingress load balancer test for None and vSphere platforms as it has to be created manually
```
Comment (selansen) [inline test/e2e/network_test.go:234]: Can you pls add back the below comment from the delete part? Not sure if this is going to be useful but  don't want to miss out the link if it is useful.

File: test/e2e/network_test.go [hunk 3]
```diff
@@ -233,6 +231,12 @@ func testNorthSouthNetworking(t *testing.T) {
...
+	// https://docs.openshift.com/container-platform/4.5/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-load-balancer.html#nw-using-load-balancer-getting-traffic_configuring-ingress-cluster-traffic-load-balancer
+	if testCtx.CloudProvider.GetType() == config.VSpherePlatformType ||
+		testCtx.CloudProvider.GetType() == config.NonePlatformType {
+		t.Skipf("NorthSouthNetworking test is disabled for platform %s", testCtx.CloudProvider.GetType())
```
Comment (alinaryan) [inline test/e2e/network_test.go:238]: Can you edit the commit message to say you are using t.Skipf?

Comment (jrvaldes) [reply test/e2e/network_test.go:238]: Good catch!

Fixed.

**Comment by sebsoto** [comment]
@jrvaldes please run the platform none test on this PR to ensure the correct behavior is being seen

**Comment by jrvaldes** [comment]
> @jrvaldes please run the platform none test on this PR to ensure the correct behavior is being seen

For the test to pass needs https://github.com/openshift/windows-machine-config-operator/pull/900 and https://github.com/openshift/windows-machine-config-operator/pull/899

**Comment by sebsoto** [comment]
> For the test to pass needs https://github.com/openshift/windows-machine-config-operator/pull/900 and https://github.com/openshift/windows-machine-config-operator/pull/899

The test doesn't need to fully pass it just needs to get pass the issue being fixed in this PR

**Comment by jrvaldes** [comment]
/test platform-none-vsphere-e2e-operator

Rebased with https://github.com/openshift/windows-machine-config-operator/pull/900

**Comment by jrvaldes** [comment]
WMCO's e2e test suite [passed](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/896/pull-ci-openshift-windows-machine-config-operator-master-platform-none-vsphere-e2e-operator/1496899128838328320) in an UPI platform-none cluster.  :tada: 

The job [failed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/896/pull-ci-openshift-windows-machine-config-operator-master-platform-none-vsphere-e2e-operator/1496899128838328320/artifacts/platform-none-vsphere-e2e-operator/upi-deprovision-vsphere/build-log.txt) in the upi-deprovision-vsphere step.

Opened https://github.com/openshift/release/pull/26558 to fix the order of the deprovision steps.

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

**Comment by jrvaldes** [comment]
/hold

for Azure CI [failure](https://coreos.slack.com/archives/C0VMT03S5/p1645772057712409): failed to initialize the cluster: Cluster operator monitoring is not available

File: test/e2e/network_test.go [hunk 4]
```diff
@@ -233,6 +231,12 @@ func testNorthSouthNetworking(t *testing.T) {
 	testCtx, err := NewTestContext()
 	require.NoError(t, err)
 
+	// Ignore the application ingress load balancer test for None and vSphere platforms as it has to be created manually
+	// https://docs.openshift.com/container-platform/4.5/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-load-balancer.html#nw-using-load-balancer-getting-traffic_configuring-ingress-cluster-traffic-load-balancer
```
Comment (sebsoto) [inline test/e2e/network_test.go:235]: lets not link to 4.5 docs here, updating it to 4.9 would be preferred if 4.10 docs arent up yet

File: test/e2e/delete_test.go [hunk 5]
```diff
@@ -115,7 +115,10 @@ func testWindowsNodeDeletion(t *testing.T) {
 	testCtx, err := NewTestContext()
 	require.NoError(t, err)
 
+	// set excepted node count to zero, since all Windows Nodes should be deleted in the test
```
Comment (sebsoto) [inline test/e2e/delete_test.go:118]: excepted -> expected

File: test/e2e/delete_test.go [hunk 6]
```diff
@@ -115,7 +115,10 @@ func testWindowsNodeDeletion(t *testing.T) {
...
+	expectedNodeCount := int32(0)
 	// Get all the Machines created by the e2e tests
+	// For platform=none, the resulting collection is empty
```
Comment (sebsoto) [inline test/e2e/delete_test.go:121]: collection -> slice

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: test/e2e/network_test.go (hunk 4 above)
Comment (saifshaikh48) [reply test/e2e/network_test.go]: Another suggestion is replacing an explicit version with `latest` so it's always up to date

Comment (jrvaldes) [reply test/e2e/network_test.go]: I'd rather stick with `4.9`. The content may change in the future and there is a possibility that the `latest` documentation is no longer relevant to this context.

Comment (jrvaldes) [reply test/e2e/network_test.go]: IMO, the link points to general documentation not necessarily related to this issue and should be removed. I [added](https://github.com/openshift/windows-machine-config-operator/pull/896#discussion_r806338505) it back per @selansen  request.

**Pull request merged**